presto-main/src/main/java/com/facebook/presto/connector/system/SystemConnectorSessionUtil.java

```diff
@@ -34,7 +34,7 @@ public static Session toSession(ConnectorTransactionHandle transactionHandle, Co
     {
         TransactionId transactionId = ((GlobalSystemTransactionHandle) transactionHandle).getTransactionId();
         ConnectorIdentity connectorIdentity = session.getIdentity();
-        Identity identity = new Identity(connectorIdentity.getUser(), connectorIdentity.getPrincipal());
+        Identity identity = new Identity(connectorIdentity.getUser(), connectorIdentity.getPrincipal(), connectorIdentity.getExtraCredentials());
```

Comment:
I'm not very familiar with this code, but I'm wondering if we should just populate the entire `Identity` object? There are fields within `Identity` that are the same as in `ConnectorIdentity` such as `extraAuthenticators`, `selectedUser`, and `reasonForSelect`. Should we also populate those too or would there be adverse side effects?

Reply:
In this case, populating only the necessary fields in the Identity object should be sufficient. Since my goal is to ensure that the system connector (e.g., system.jdbc.tables) works across connectors like the Arrow Flight connector without failures, I focused on fields that are required for this functionality to work correctly.

Reply:
I was thinking like, If the system connector doesn't explicitly require fields like extraAuthenticators, selectedUser, or reasonForSelect, there is no need to pass them. These fields could typically be used in more specific cases, such as when additional authentication mechanisms are involved, or when user impersonation is necessary.

extraAuthenticators: This is used when additional authentication layers are needed. If the system connector or your use case doesn’t involve extra authentication, passing this field is unnecessary.

selectedUser: This is mainly relevant for scenarios where user impersonation or proxy users are involved. If your system connector doesn’t handle such cases, it’s safe to omit this field.

reasonForSelect: This field provides context for why a user switch occurred.

Reply:
Thanks for the context. I think it's fine to leave the additional fields out for now

Reply:
When you have a chance, could you approve the changes if everything looks good? @ZacBlanco